### PR TITLE
(PUP-10169) Generalize forcelocal parameter for user/group

### PR DESCRIPTION
**Pull request by luchihoratiu**

### CODE REVIEW

File: lib/puppet/feature/manages_local_users_and_groups.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require 'puppet/util/feature'
+require 'puppet/util/libuser'
+
+Puppet.features.add(:manages_local_users_and_groups) {
+  Puppet.features.libuser? || Facter.value(:kernel) == "AIX"
```
Comment (ciprianbadescu) [inline lib/puppet/feature/manages_local_users_and_groups.rb]: Is this needed as the feature looks to be enabled only on aix and libuser cases already?

File: lib/puppet/util/libuser.rb [hunk 2]
```diff
@@ -8,5 +8,4 @@ def self.getenv
      newenv['LIBUSER_CONF'] = getconf
      newenv
    end
-
```
Comment (ciprianbadescu) [inline lib/puppet/util/libuser.rb]: nit, file should not be added into commit for this change

File: lib/puppet/feature/manages_local_users_and_groups.rb (hunk 1 above)
Comment (luchihoratiu) [reply lib/puppet/feature/manages_local_users_and_groups.rb]: It seems it is not needed anymore.

File: spec/unit/type/user_spec.rb [hunk 3]
```diff
@@ -408,7 +408,6 @@ def self.instances; []; end
       Puppet::Resource::Catalog.new :testing do |conf|
         [testuser, testrole].each { |resource| conf.add_resource resource }
       end
-      allow(Puppet::Type::User::ProviderDirectoryservice).to receive(:get_macosx_version_major).and_return("10.5")
```
Comment (ciprianbadescu) [inline spec/unit/type/user_spec.rb]: how is this change linked to 'forcelocal'?

Comment (GabrielNagy) [reply spec/unit/type/user_spec.rb]: this is a spec file that failed on OS X because that function does not exist anymore... it's not really related

**Pull request merged**